clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/ArraysTests.scala

```diff
@@ -0,0 +1,1040 @@
+        (fingerprintingSummaryMetricsSource, fingerprintingSummaryMetricsContents)
+      ).foreach {
+        case (destination, contents) =>
+          destination.contentAsString should be(contents)
+      }
+      succeed
+    }
+
+    result.andThen {
+      case _ =>
+        val _ = Seq(
+          vcfSource,
+          vcfDestination,
+          vcfIndexSource,
+          craiDestination,
+          fingerprintingDetailMetricsSource,
+          alignmentMetricsDestination,
+          fingerprintingSummaryMetricsSource,
+          fingerprintMetricsDestination
+        ).map(_.delete(swallowIOExceptions = true))
+    }
+  }
+
+  it should "move the vcf and vcfIndex together in GCP" in testMove()
+
+  it should "support changing the vcf and vcfIndex basename on move" in testMove(
+    changeBasename = true
+  )
+
+  it should "not move arrays without a destination" in {
+    recoverToExceptionIf[Exception] {
+      runClient(
+        ClioCommand.moveArraysName,
+        "--location",
+        Location.GCP.entryName,
+        "--chipwell-barcode",
+        randomId,
+        "--version",
+        "123"
+      )
+    }.map {
+      _.getMessage should include("--destination")
+    }
+  }
+
+  it should "not move arrays with no registered files" in {
+    val key = ArraysKey(
+      Location.GCP,
+      Symbol(s"barcode$randomId"),
+      1
+    )
+    runUpsertArrays(key, ArraysMetadata()).flatMap { _ =>
+      recoverToExceptionIf[Exception] {
+        runClient(
+          ClioCommand.moveArraysName,
+          "--location",
+          key.location.entryName,
+          "--chipwell-barcode",
+          key.chipwellBarcode.name,
+          "--version",
+          key.version.toString,
+          "--destination",
+          "gs://some-destination/"
+        )
+      }.map {
+        _.getMessage should include("Nothing to move")
+      }
+    }
+  }
+
+  def testDelete(
+    existingNote: Option[String] = None,
+    testNonExistingFile: Boolean = false,
+    force: Boolean = false,
+    workspaceName: Option[String] = None
+  ): Future[Assertion] = {
+    val deleteNote = s"$randomId --- Deleted by the integration tests --- $randomId"
+
+    val id = randomId
+    val barcode = s"barcode$id"
+    val version = 3
+
+    val vcfContents = s"$id --- I am a vcf fated to die --- $id"
+    val vcfIndexContents = s"$id --- I am an index fated to die --- $id"
+    val metrics1Contents = s"$id --- I am an immortal metrics file --- $id"
+    val metrics2Contents =
+      s"$id --- I am a second immortal metrics file --- $id"
+
+    val storageDir = rootTestStorageDir / s"arrays/$barcode/v$version/"
+    val vcfPath = storageDir / s"$randomId${ArraysExtensions.VcfGzExtension}"
+    val vcfIndexPath = storageDir / s"$randomId${ArraysExtensions.VcfGzTbiExtension}"
+    val metrics1Path = storageDir / s"detail-$id.metrics"
+    val metrics2Path = storageDir / s"summary-$id.metrics"
```

Comment:
we should also probably assert that all files that are not supposed to be deleted are left alone.

Reply:
I think that's what
```
        Seq((metrics1Path, metrics1Contents), (metrics2Path, metrics2Contents)).foreach {
          case (path, contents) =>
            path should exist
            path.contentAsString should be(contents)
        }
```
below is doing